code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -112,6 +112,8 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		var/mob/living/carbon/human/C = loc
 		if(istype(C) && src == C.wear_mask)
 			reagents.trans_to_mob(C, burn_rate*initial_volume, CHEM_BREATHE, 0.75)
+			if(C.check_has_mouth() && prob(5))
+				reagents.trans_to_mob(C, burn_rate*initial_volume, CHEM_INGEST, 0.75)

```

Comment:
If it isn't probability, it procs on every tick. Every single one. Because it's probability, you're only reminded you're tasting it about once per minute (by my testing). Because of that, simply adding it on to breathing would be incredibly obtrusive.

Reply:
You need to combine probability with a sanity check, or better yet just a check to see if it’s the first time we’re imbibing the substance and if so display the message. Regardless that still does not justify the, imo incredibly hacky, method you’re currently employing of both calling transfer vis CHEM_BREATH and via CHEM_INGEST - it shoukd just be the former.

Reply:
IPCs should not be able to inhale reagents in the first place.

Reply:
That... unfortunately isn't the purview of this PR. Not letting them inhale at all sounds like a change to IPCs. This PR just makes it so that this happens over the course of 5 minutes: https://puu.sh/CqW8N/ad07d30cca.png

Now that first proc doesn't always occur, it just happened rather serendipitously when I just retested it. Any tips on making sure it happens on the first instance would be appreciated!

Reply:
Then adjust the purview of your PR. If you are trying to implement something and in the course of doing so you encounter an obstacle that can either be circumvented with a hackt solution or fixed with a thorough solution we do not opt for the hacky solution just because the thorough solution is “outside the original intended scope of the peoject” aka “I’m too lazy to do that”.

Reply:
It isn't that I'm too lazy, it's that I legitimately don't know how to accomplish it. If we make it so that CHEM_BREATHE gives you taste as well as inhaling and make it so that IPCs aren't capable of CHEM_BREATHE this poses a few problems. First and foremost, every time CHEM_BREATHE occurs, you'll get a taste message in the chat. That's way too often and oppressive. We'd have to find a way to make it not give you a taste message despite the fact we've just made CHEM_BREATHE give you taste.  Secondly, this brings up the other problem IPCs smoking cigarettes near-infinitely. Cigarettes don't actually last that long even if you aren't actively inhaling. 

I've taken CHEM_BREATHE in this situation to be as if you're lazily puffing away occasionally. CHEM_INGEST would be if you take a long drag of the cigarette. 

I'd love to solve every problem. But I do not have the skills to do so. I came across this and thought it would be an interesting thing to add because there's no difference between smoking cigarette brands right now. I'd love to make it better, but I'd need guidance other than "this is a hacky solution" and "You're lazy".